src/librustc/mir/interpret/mod.rs

```diff
@@ -575,6 +575,144 @@ impl<Tag, Extra: Default> Allocation<Tag, Extra> {
             extra: Extra::default(),
         }
     }
+
+    #[inline]
+    pub fn size(&self) -> Size {
+        Size::from_bytes(self.bytes.len() as u64)
+    }
+
+    pub fn check_align(
+        &self,
+        offset: Size,
+        required_align: Align,
+    ) -> EvalResult<'tcx> {
+        if self.align.abi() > required_align.abi() {
```

Comment:
Isn't this the wrong way around? If we are more aligned, that's good, right?

Reply:
already fixed in this PR in a commit further down